VMEncryption/main/DiskUtil.py

```diff
@@ -743,6 +745,30 @@ def query_dev_id_path_by_sdx_path(self, sdx_path):
 
         return sdx_path
 
+    def get_persistent_path_by_sdx_path(self, sdx_path):
+        """
+        return a stable path for this /dev/sdx device
+        """
+        sdx_realpath = os.path.realpath(sdx_path)
+
+        # First try finding an Azure symlink
+        azure_name_table = self.get_block_device_to_azure_udev_table()
+        if sdx_realpath in azure_name_table:
+            return azure_name_table[sdx_realpath]
+
+        # Then try matching a uuid symlink. Those are probably the best
+        for disk_by_uuid in os.listdir(CommonVariables.disk_by_uuid_root):
+            disk_by_uuid_path = os.path.join(CommonVariables.disk_by_uuid_root, disk_by_uuid)
+
+            if os.path.realpath(disk_by_uuid_path) == sdx_realpath:
+                return disk_by_uuid_path
+
+        # Found nothing very persistent. Just return the original sdx path.
+        # And Log it.
+        self.logger.log(msg="Failed to find a persistent path for [{0}].".format(sdx_path), level=CommonVariables.WarningLevel)
+
```

Comment:
Do we expect this condition to occur? I would recommend to fail here rather than silently proceed with the dev path (/dev/sdd1 or something). We will end up failing to mount the data disks later on.

Reply:
This can occur when they are encrypting over RAID or LVM. But that is fine because mapper names arising from those systems are persistent.
That is why I don't want to just error out there.